contracts/upgradeable_contracts/components/common/SelectorTokenGasLimitManager.sol

```diff
@@ -0,0 +1,113 @@
+pragma solidity 0.7.5;
+
+import "../../BasicAMBMediator.sol";
+
+/**
+ * @title SelectorTokenGasLimitManager
+ * @dev Basic storage and methods needed by mediators to interact with AMB bridge.
+ * @dev Functionality for determining the request gas limit for AMB execution based on the request selectors and bridged tokens.
+ */
+abstract contract SelectorTokenGasLimitManager is BasicAMBMediator {
```

Comment:
Is this manager is a good candidate to be in a separate contract to reduce bytecode of the OB contract?

Reply:
I have checked such possibility, the size of bytecode got reduced only by ~200 bytes. For now, I think there is no much sense in doing so.

Reply:
Update:
After adding heavy helper functions, it started to make sense to extract functionality to the separate module. So, I added a new Home OB module for that.